tools/donate-cpu-server.py

```diff
@@ -1268,16 +1277,18 @@ def server(server_address_port: int, packages: list, packageIndex: int, resultPa
         try:
             bytes_received = connection.recv(128)
             cmd = bytes_received.decode('utf-8', 'ignore')
-        except socket.error:
+        except socket.error as e:
+            print_ts('Error: Recv error: ' + str(e))
             connection.close()
             continue
         except UnicodeDecodeError as e:
-            connection.close()
             print_ts('Error: Decoding failed: ' + str(e))
+            connection.close()
             continue
         pos_nl = cmd.find('\n')
         if pos_nl < 1:
             print_ts('No newline found in data.')
+            connection.close()
```

Comment:
Coincidentally we are hitting this error path a lot because of an issue so we are probably exhausting the connections on the system. See https://trac.cppcheck.net/ticket/12669.

Reply:
This was caused by browser changing the protocol to `https://` leading to this error path being reached which did not close the connection so essentially DoS'ing the server.